runtime/hydradx/src/system.rs

```diff
@@ -305,9 +310,32 @@ impl pallet_utility::Config for Runtime {
 	type RuntimeEvent = RuntimeEvent;
 	type RuntimeCall = RuntimeCall;
 	type PalletsOrigin = OriginCaller;
+	type BatchPreHook = PushBatchExecutionTypeForUnifiedEvent;
+	type BatchPostHook = PopBatchExecutionTypeForUnifiedEvent;
 	type WeightInfo = weights::pallet_utility::HydraWeight<Runtime>;
 }
 
+pub struct PushBatchExecutionTypeForUnifiedEvent;
+
+impl BatchPreHook for PushBatchExecutionTypeForUnifiedEvent {
+	fn on_batch_start() -> DispatchResult {
+		let next_event_id = AmmSupport::next_id().map_err(|_| ArithmeticError::Overflow)?;
+		AmmSupport::push(ExecutionType::Batch(next_event_id))?;
```

Comment:
here we can clearly see why I added my previous "nope" comment ( where it implements the same interface for Pallet).

first point:
If you try to use interface of AmmSupport - you will get these options next_id(), next_increment_id(), get() (some id)  and probabyl few others as well ( from the storage gettters).

since there seem to be 2 types of ids afaik - it becomes very messy and unclear what to actually use.

second point - why the next_id and push has to be outside the support pallet - it is a logic that should be hidden inside the pallet imo.